upi/azure/azuredeploy.json

```diff
@@ -0,0 +1,1072 @@
+{
+  "$schema" : "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion" : "1.0.0.0",
+  "parameters" : {
+    "image" : {
+      "type" : "string",
+      "metadata" : {
+        "description" : "OS to use.Red Hat Enterprise Linux"
+      }
+    },
+    "BootstrapIgnition" : {
+      "type" : "string",
+      "minLength" : 1,
+      "metadata" : {
+        "description" : "User name for the Virtual Machine and OpenShift Webconsole."
+      }
+    },
+    "MasterIgnition" : {
+      "type" : "string",
+      "metadata" : {
+        "description" : "User password for the OpenShift Webconsole"
+      }
+    },
+    "WorkerIgnition" : {
+      "type" : "string",
+      "metadata" : {
+        "description" : "User password for the OpenShift Webconsole"
+      }
+    },
+    "numberOfNodes" : {
+      "type" : "int",
+      "defaultValue" : 3,
+      "minValue" : 2,
+      "maxValue" : 30,
+      "metadata" : {
+        "description" : "Number of OpenShift Nodes to deploy (max 30)"
+      }
+    },
+    "sshKeyData" : {
+      "type" : "securestring",
+      "metadata" : {
+        "description" : "SSH RSA public key file as a string."
+      }
+    },
+    "masterVMSize" : {
```

Comment:
we require 4 cores, 16 gb for control-plane as minimum requirements. So probably only allow those.